Sources/ConnectionPoolModule/PoolStateMachine+ConnectionGroup.swift

```diff
@@ -449,6 +449,21 @@ extension PoolStateMachine {
             return (index, context)
         }
 
+        /// Returns `true` if the connection should be explicitly closed, `false` if nothing needs to be done.
+        @inlinable
+        mutating func keepAliveFailed(_ connectionID: Connection.ID) -> Bool {
```

Comment:
Can we return an enum for this? KeepAliveFailedAction with two cases: close and none? Makes this more explicit

Reply:
This is now returning an optional CloseAction. It should be explicit enough now, let me know if not.